app/livechat/server/api/lib/visitors.js

```diff
@@ -41,3 +41,34 @@ export async function findVisitedPages({ userId, roomId, pagination: { offset, c
 		total,
 	};
 }
+
+export async function findChatHistory({ userId, roomId, visitorId, pagination: { offset, count, sort } }) {
+	if (!await hasPermissionAsync(userId, 'view-l-room')) {
+		throw new Error('error-not-authorized');
+	}
+	const room = await LivechatRooms.findOneById(roomId);
+	if (!room) {
+		throw new Error('invalid-room');
+	}
+	const subscription = Subscriptions.findOneByRoomIdAndUserId(room._id, this.userId, { fields: { _id: 1 } });
```

Comment:
Is this to check permission or something like this?
If so, it won't work right, because the `livechat managers` have access to the `rooms` even when they don't have the subscription, you know?
In this case, we use the `canAccessRoom` method to check permission.